File: exercises/java/src/test/java/checkout/ReceiptTests.java
package checkout;

import org.junit.Test;

import static org.assertj.core.api.Assertions.*;

public class ReceiptTests {

    @Test
    public void singleItem() {
        Checkout checkout = new Checkout();
        checkout.scan("A");
        assertThat(checkout.receipt()).containsSequence(
                "A: 50\n",
                "Total: 50");
    }
    
    @Test
    public void oneOfEach() {
        Checkout checkout = new Checkout();
        checkout.scan("A");
        checkout.scan("B");
        checkout.scan("C");
        checkout.scan("D");
        assertThat(checkout.receipt()).containsSequence(
                "A: 50\n",
                "B: 30\n",
                "C: 20\n",
                "D: 15\n",
                "Total: 115");
    }

    @Test
    public void offers() {
        Checkout checkout = new Checkout();
        checkout.scan("A");
        checkout.scan("A");
        checkout.scan("B");
        checkout.scan("A");
        checkout.scan("C");
        checkout.scan("D");
        checkout.scan("B");
        assertThat(checkout.receipt()).containsSequence(
                "A: 50\n",
                "A: 50\n",
                "B: 30\n",
                "A: 50 - 20 (3 for 130)\n",
                "C: 20\n",
                "D: 15\n",
                "B: 30 - 15 (2 for 45)\n",
                "Total: 210");
    }
}

File: exercises/dotnet/McrDigital.Bootcamp1.Checkout.Tests/ReceiptTests.cs
namespace McrDigital.Bootcamp1.Checkout.Tests {
  using System;
  using Xunit;
  using McrDigital.Bootcamp1.Checkout;

  public class ReceiptTests {
    [Fact]
    public void SingleItem() {
      var checkout = new Checkout();
      checkout.Scan("A");
      Assert.Equal(new string[] {
        "A: 50",
        "Total: 50"
      }, checkout.Receipt.Split("\n"));
    }

    [Fact]
    public void OneOfEach() {
      var checkout = new Checkout();

      checkout.Scan("A");
      checkout.Scan("B");
      checkout.Scan("C");
      checkout.Scan("D");

      Assert.Equal(new string[] {
        "A: 50",
        "B: 30",
        "C: 20",
        "D: 15",
        "Total: 115"
      }, checkout.Receipt.Split("\n"));
    }

    [Fact]
    public void Offers() {
      var checkout = new Checkout();

      checkout.Scan("A");
      checkout.Scan("A");
      checkout.Scan("B");
      checkout.Scan("A");
      checkout.Scan("C");
      checkout.Scan("D");
      checkout.Scan("B");

      Assert.Equal(new string[] {
        "A: 50",
        "A: 50",
        "B: 30",
        "A: 50 - 20 (3 for 130)",
        "C: 20",
        "D: 15",
        "B: 30 - 15 (2 for 45)",
        "Total: 210"
      }, checkout.Receipt.Split("\n"));
    }
  }
}